File: ported/auto/4110f6.js
// Auto-translated from Ghidra C by tools/c-to-js/translate.js.
// Source: decompiled/c/4110f6.c
// Edit by hand only after diff-test passes — re-running the translator will overwrite.

/** @typedef {import("../runtime/heap.js").Heap} Heap */

import { FUN_00410fc7 } from "./410fc7.js";
import { FUN_0041107f } from "./41107f.js";
export function FUN_004110f6(heap, param_1) {
  let iVar1 = 0;
  let bVar2 = 0;
  if (heap.u32(0x005ec158) == 0x0) {
    bVar2 = false;
  } else {
    FUN_0041107f(heap);
    if ((heap.u32(0x005ec160) == 0) && (param_1 != 0)) {
      iVar1 = (heap.u32(heap.u32((heap.u32(heap.u32(0x005ec158)) + 0x34))))(heap.u32(0x005ec158), heap.u32((param_1 + 0x104)), FUN_00410fc7, 0, 0x80);
      bVar2 = iVar1 == 0;
    } else {
      iVar1 = (heap.u32(heap.u32((heap.u32(heap.u32(0x005ec158)) + 0x34))))(heap.u32(0x005ec158), heap.u32(0x005f0320), FUN_00410fc7, 0, 0);
      bVar2 = iVar1 == 0;
    }
  }
  return bVar2;
}

File: decompiled/c/4110f6.c

bool FUN_004110f6(int param_1)

{
  int iVar1;
  bool bVar2;
  
  if (DAT_005ec158 == (int *)0x0) {
    bVar2 = false;
  }
  else {
    FUN_0041107f();
    if ((DAT_005ec160 == 0) && (param_1 != 0)) {
      iVar1 = (**(code **)(*DAT_005ec158 + 0x34))
                        (DAT_005ec158,*(undefined4 *)(param_1 + 0x104),FUN_00410fc7,0,0x80);
      bVar2 = iVar1 == 0;
    }
    else {
      iVar1 = (**(code **)(*DAT_005ec158 + 0x34))(DAT_005ec158,DAT_005f0320,FUN_00410fc7,0,0);
      bVar2 = iVar1 == 0;
    }
  }
  return bVar2;
}

